src/hooks/useBeforeUnload.js

```diff
@@ -1,19 +1,39 @@
-import React, { useEffect } from 'react';
+import { useEffect } from 'react';
 
-export default function useBeforeunload ({ contentIsDirty, handleClickOpen }) {
-    useEffect(() => {
-        const handleBeforeunload = (event) => {
-          event.preventDefault()
-          handleClickOpen()
-        }
-    
-        if (!contentIsDirty) {
-          window.addEventListener('beforeunload', handleBeforeunload)
-        }
-    
-        return () => {
-          window.removeEventListener('beforeunload', handleBeforeunload)
-        }
-      }, [contentIsDirty])
+export default function useBeforeunload({
+  prompt,
+  isConfirmed,
+  contentIsDirty = false,
+}) {
+  console.log({ contentIsDirty });
+  useEffect(() => {
+    const handleBeforeunload = (event) => {
+      event.preventDefault();
+
+      // const confirm = await isConfirmed(prompt);
+      // if (confirm) {
+      //   event.preventDefault();
+      // }
+
+      // Chrome requires `returnValue` to be set.
+      // if (event.defaultPrevented) {
+      //   event.returnValue = '';
+      // }
+
+      // if (typeof returnValue === 'string') {
+      //   event.returnValue = 'promptText';
+      //   return 'promptText';
+      // }
```

Comment:
will remove the comented parts once i will merge and test in my local before raising PR against develop